example/example.h

```diff
@@ -0,0 +1,15 @@
+#ifndef GROUP_EXAMPLE_HEADER_GUARD
+#define GROUP_EXAMPLE_HEADER_GUARD
+
+#include <stdint.h>
+#include <stddef.h>
+
+
+void set_function_index_group_example(void** functions);
+
+int add(int a, int b);
+
+void print_factorial(int n);
+
+
+#endif /* GROUP_EXAMPLE_HEADER_GUARD */
```

Comment:
It is a good habit to write a comment indicating which `#if` does this `#endif` match (also applicable to `#else`). It's great that you've picked it up.

I just want to mention that you could also have written
```suggestion
#endif // GROUP_EXAMPLE_HEADER_GUARD
```
Since `//` comments are also part of C since C99.